File: python/applied_enumeration.py
from enum import Enum
import random

class GameConstants(Enum):
    PLAYER = 100
    COM = 100

class MainApplication:
    def mainFunction():

        playerHealth = GameConstants.PLAYER.value
        comHealth = GameConstants.COM.value

        #healing and dodge
        heal = 20
        playerDodgeToken = 0
        comDodgeToken = 0

        #damage negation
        damageNegated = None
        playerDamageRolled = None
        comDamageRolled = None
        damageDoneWithNegation = None

        print("Welcome to a short turn base game, prepare to battle against a COM")

        #start loop
        while(True):
            print(f"-------------------------------------------------------")
            print(f"Please type in 1, 2 or 3 to make your move")
            print(f"1 ATTACK")
            print(f"2 HEAL")
            print(f"3 DODGE")
            print(f"-------------------------------------------------------")

            #showing Player Health and COM Health
            print(f"Player Health: ", playerHealth)
            print(f"COM Health: ", comHealth)

            #start of the player turn
            playerTurn = int(input())

            if(playerTurn == 1):

                if(comDodgeToken >= 1):
                    
                    damageNegated = random.randrange(10, 20)
                    playerDamgeRolled = random.randrange(10, 30)

                    if(damageNegated >= playerDamgeRolled):

                        print(f"The COM took no damge because of his dodge token")
                        print(f"")
                        comDodgeToken = 0

                    else:
                        damageDoneWithNegation = abs(damageNegated - playerDamageRolled)
                        print(f"The COM rolled a", damageNegated, "damaged negation you did", damageDoneWithNegation)
                        print(f"")
                        comHealth -= damageDoneWithNegation
                        comDodgeToken = 0
                else:
                    playerDamageRolled = random.randrange(10, 30)
                    comHealth -= playerDamageRolled
                    print(f"You did", playerDamageRolled, "damage")
                    print(f"")

            elif(playerTurn == 2):

                if(playerHealth >= 100):
                    print(f"You cannot heal passed 100")
                    print(f"")

                else:

                    if(playerHealth + heal >= 100):

                        playerHealth = 100
                        print(f"You have healed to 100")
                        print(f"")
                    
                    else:

                        playerHealth += heal
                        print(f"You healed", heal, "health")
                        print(f"")
            
            elif(playerTurn == 3):

                if(playerDodgeToken >= 1):
                    print(f"You can only have one dodgetoken at a time")
                    print(f"")
                else:
                    playerDodgeToken+=1
                    print(f"You have acquired a dodge token ")
                    print(f"")
        
            #start of the COM turn
            computerTurn = random.randrange(1,3)

            if(computerTurn == 1):

                if(playerDodgeToken >= 1):

                    damageNegated = random.randrange(10,20)
                    comDamageRolled = random.randrange(10,20)

                    if(damageNegated >= comDamageRolled):

                        print(f"The player took no damage because of his dodge token")
                        print(f"")
                        playerDodgeToken = 0
                    
                    else:

                        damageDoneWithNegation = abs(damageNegated - comDamageRolled)
                        print(f"The Player rolled a", damageNegated, "damaged negation COM did", damageDoneWithNegation)
                        playerHealth -= damageDoneWithNegation
                        playerDodgeToken = 0
                else:
                    comDamageRolled = random.randrange(10,20)
                    playerHealth -= comDamageRolled
                    print(f"The COM did", comDamageRolled, "damge")
                    print(f"")

            elif(computerTurn == 2):

                if(comHealth >= 100):

                    print(f"You cannot healh passed 100")
                    print(f"")
                else:

                    if(comHealth + heal >= 100):

                        comHealth = 100
                        print(f"The COM has healed to 100")
                        print(f"")

                    else:

                        comHealth += heal
                        print(f"The COM has healed", heal, "health")
                        print(f"")

            elif(computerTurn == 3):

                if(comDodgeToken >= 1):

                    print(f"The COM can only have one dodge token at a time")
                    print(f"")

                else:

                    comDodgeToken+=1
                    print(f"The COM has acquired a dodge token")
                    print(f"")

            if(comHealth <= 0):
                print(f"You have won the game")
                break
            elif(playerHealth <= 0):
                print(f"The COM has won")
                break    



        
if __name__ == "__main__":
    MainApplication.mainFunction()
    

# https://www.geeksforgeeks.org/python/python-random-module/
# https://docs.python.org/3/library/random.html

File: java/AppliedEnumeration.java
import java.util.Random;
import java.util.Scanner;

class AppliedEnumeration {
    public static enum GameConstants {

        PLAYER("Player", 100),
        COM("Computer", 10)
        ;

        private String name;
        private int health;

        GameConstants(String n, int h){
            this.name = n;
            this.health = h;
        }
    }
    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        //Random numbers
        Random comTurn = new Random();
        Random damageChance = new Random();
        Random dodgeChacne = new Random();
        
        //healing and dodge
        int heal = 20;
        int playerDodgeToken = 0;
        int comDodgeToken = 0;

        //damage negation
        int damageNegated;
        int playerDamageRolled;
        int comDamageRolled;
        int damgeDoneWithNegation;

        System.out.println("Welcome to a short turn base and prepare battle against a COM");

        //start of the game loop
        while(true){
            System.out.println("-------------------------------------------------------");
            System.out.println("Please type in 1, 2 or 3 to make your move: ");
            System.out.println("1 ATTACK");
            System.out.println("2 HEAL");
            System.out.println("3 DODGE");
            System.out.println("-------------------------------------------------------");

            //showing Player Health and COM Health
            System.out.println();
            System.out.println("Player Health: " + GameConstants.PLAYER.health);
            System.out.println("COM Health: " + GameConstants.COM.health);

            //start of the player turn
            int playerTurn = s.nextInt();

            if(playerTurn == 1){

                if(comDodgeToken >= 1) {

                    damageNegated = dodgeChacne.nextInt(10, 20);
                    playerDamageRolled = damageChance.nextInt(10,31);

                    if(damageNegated >= playerDamageRolled){

                        System.out.println("The COM took no damage because of his dodge token");
                        comDodgeToken = 0;
                    } else {

                        damgeDoneWithNegation = Math.abs(damageNegated - playerDamageRolled);
                        System.out.println("The COM rolled a " + damageNegated + " damaged negation " + "you did " + damgeDoneWithNegation);
                        GameConstants.COM.health -= damgeDoneWithNegation;
                        comDodgeToken = 0;
                    }
                } else {

                    playerDamageRolled = damageChance.nextInt(10, 31);
                    GameConstants.COM.health -= playerDamageRolled;
                    System.out.println("You did " + playerDamageRolled + " damage");
                    System.out.println();
                }
            } else if(playerTurn == 2){

                if(GameConstants.PLAYER.health >= 100){
                    System.out.println("You cannot heal passed 100");
                    System.out.println();
                } else {

                    if(GameConstants.PLAYER.health + heal >= 100){

                        GameConstants.PLAYER.health = 100;
                        System.out.println("You have healed to 100");
                        System.out.println();

                    } else {

                        GameConstants.PLAYER.health += heal;
                        System.out.println("You healed " + heal + " health");
                        System.out.println();
                    }
                }
            
            } else if(playerTurn == 3) {

                if(playerDodgeToken >= 1){
                    System.out.println("You can only have one dodge token at a time");
                    System.out.println();
                } else {
                    playerDodgeToken++;
                    System.out.println("You have acquired a dodge token");
                    System.out.println();
                }
            }


            //start of the COM Turn
            int computerTurn = comTurn.nextInt(1,4);

            if(computerTurn == 1){

                if(playerDodgeToken >= 1){
                    damageNegated = dodgeChacne.nextInt(10, 20);
                    comDamageRolled = damageChance.nextInt(10, 31);

                    if(damageNegated >= comDamageRolled){

                        System.out.println("The Player took no damage because of his dodge token");
                        playerDodgeToken = 0;
                    } else {

                        damgeDoneWithNegation = Math.abs(damageNegated - comDamageRolled);
                        System.out.println("The Player rolled a " + damageNegated + " damaged negation " + "COM did " + damgeDoneWithNegation);
                        GameConstants.PLAYER.health -= damgeDoneWithNegation;
                        playerDodgeToken = 0;
                    }
                } else {

                    comDamageRolled = damageChance.nextInt(10, 31);
                    GameConstants.PLAYER.health -= comDamageRolled;
                    System.out.println("The COM did " + comDamageRolled + " damage \n");
                    System.out.println();
                }

            } else if(computerTurn == 2){

                if(GameConstants.COM.health >= 100){
                    System.out.println("You cannot heal passed 100 \n");
                    System.out.println();
                } else {

                    if(GameConstants.COM.health + heal >= 100){

                        GameConstants.COM.health = 100;
                        System.out.println("The com has healed to 100");
                        System.out.println();
                    } else {

                        GameConstants.COM.health += heal;
                        System.out.println("The com has healed " + heal + " health");
                        System.out.println();
                    }
                }
            } else if(computerTurn == 3){

                if(comDodgeToken >= 1){
                    System.out.println("The com only have one dodge token at a time");
                    System.out.println();
                } else {
                    comDodgeToken++;
                    System.out.println("The com has acquired a dodge token");
                    System.out.println();
                }
            }

            //checking to see if health reaches 0
            if(GameConstants.COM.health <= 0){
                System.out.println("You have won the game");
                break;
            } else if(GameConstants.PLAYER.health <= 0){
                System.out.println("The COM has won");
                break;
            }
        }
    }

    /*
    Quick turn base game?
    Have the user enter in input 
    Enum1 for player could be attack, heal, or dodge(possibly)
        if the user picks the case 1 2 or 3 
    Enum2 for the npc random number generator for attack, heal or dodge
    */
}